### PR TITLE
feat(party): Implement create party

### PR DESCRIPTION
**Pull request by nedssoft**
#### What does this PR do?
 Creates a new party

#### Description of Task to be completed?
- ensure user input is validated
- ensure that only admin can access the route
- write test for the endpoint
- add new party to the database

#### How should this be manually tested?
- Login as an admin to get access token
- make POST request to /ap/v1/parties 
- Run **npm run test**
#### How should this be manually tested?
- git clone git@github.com:oriechinedu/Politico.git
- git checkout ft-api-create-political-party-163707820
- npm install
- on Postman login as an admin

```gherkin
Admin login credentials
Endpoint /api/v1/auth/login
Resquest Body
{
    email: admin@politico.com
    password: admin
}
```
- Copy the token returned and use as Authorization header for the next step
- Make a POST Request on /api/v1/parties
```
Request Body
{
    name: String,
    "hqAddress": String,
    "logoUrl": String
   
}
```
- npm run test

### CODE REVIEW

File: server/middlewares/AuthValidator.js [hunk 1]
```diff
@@ -77,6 +77,24 @@ class AuthValidator {
     }
     return next();
   }
+
+  static isAdmin(req, res, next) {
+    try {
+      const authorization = req.headers.authorization.split(' ')[1] || req.headers.token;
+
+      if (!authorization) {
+        return res.status(401).json({ error: true, message: 'Unauthorized' });
```
Comment (klevamane) [inline server/middlewares/AuthValidator.js]: I see that you have many unauthorized messages, Kindly give a more descriptive message

Comment (nedssoft) [reply server/middlewares/AuthValidator.js]: Okay. I'll do that now. Thanks.

File: server/controllers/PartyController.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+class PartyController {
+  /**
+   *
+   * @param {object} req - request
+   * @param {object} res - response
+   */
+  static async createParty(req, res) {
+    let party;
+    try {
+      party = await createParty(req, res);
```
Comment (klevamane) [inline server/controllers/PartyController.js:19]: I'm not sure why you are making use of the await keyword in front of createParty, I see that this function was implemented with the await keyword which has already returned the result

File: server/middlewares/AuthValidator.js (hunk 1 above)
Comment (nedssoft) [reply server/middlewares/AuthValidator.js]: I've implemented the feedback above. Thank you

File: server/middlewares/PartyValidator.js [hunk 3]
```diff
@@ -23,16 +40,16 @@ class PartyValidator {
...
-    // ));
-    const partyExists = helpers.recordExists(PartyModel, req);
-    if (partyExists) {
-      return res.status(409).json({
-        status: 409,
-        error: 'The party already exists',
-      });
-    }
+    (async () => {
+      const party = await partyExists(req.body.name);
```
Comment (klevamane) [inline server/middlewares/PartyValidator.js:44]: The same as above [here](https://github.com/oriechinedu/Politico/pull/61/files#r253825787)

File: server/models/PartyModel.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+  /**
+   * Checks if the party already exists
+   * @param {string} name of the party
+   */
+  static async partyExists(name) {
+    const client = await pool.connect();
+    let party;
+    try {
+      const text = 'SELECT * FROM parties WHERE name = $1 LIMIT 1';
```
Comment (klevamane) [inline server/models/PartyModel.js:38]: rename text to sqlQuery, to make it more descriptive. Do this for similar cases

File: server/controllers/PartyController.js (hunk 2 above)
Comment (nedssoft) [reply server/controllers/PartyController.js:19]: The previous function returns a promise, so for me to be to read the value returned, I have to await the response too, that's the reason I had to use async/await. For instance this part `party = await createParty(req, res);` returns *Promise <pending*>. So I have to await the Promise to resolve.

**Pull request merged**